homeassistant/components/contact_energy_nz/config_flow.py

```diff
@@ -0,0 +1,83 @@
+"""Config flow for Contact Energy NZ integration."""
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any
+
+import async_timeout
+from contact_energy_nz import AuthException, ContactEnergyApi
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+from .exceptions import CannotConnect, InvalidAuth
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Optional(CONF_API_KEY, "z840P4lQCH9TqcjC9L2pP157DZcZJMcr5tVQCvyx"): str,
```

Comment:
The token is never validated and not used in the config flow. Do you think we should remove it?

Reply:
good point. I had this idea that maybe users would like this to be configurable via HA UI but as it is now - line is indeed useless